pkg/services/apiserver/standalone/options/tracing.go

```diff
@@ -47,28 +48,33 @@ func (o *TracingOptions) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&o.JaegerPropagation, "grafana.tracing.jaeger.propagation", "jaeger", "Tracing Jaeger propagation specifies the text map propagation format, w3c or jaeger.")
 	fs.StringVar(&o.OTLPAddress, "grafana.tracing.otlp.address", "", "Tracing OTLP exporter destination, e.g. localhost:4317.")
 	fs.StringVar(&o.OTLPPropagation, "grafana.tracing.otlp.propagation", "w3c", "Tracing OTLP propagation specifies the text map propagation format, w3c or jaeger.")
+	fs.StringVar(&o.ServiceName, "grafana.tracing.service-name", "grafana-apiserver", "Override the default service name, grafana-apiserver.")
```

Comment:
❤️

Reply:
This is in general fine with me, but I'm wondering if setting [`OTEL_SERVICE_NAME`](https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/#general-sdk-configuration) works as expected here too (and which takes precedence?)

Reply:
Let me try that to verify.

Reply:
Seems like `OTEL_SERVICE_NAME` doesn't work. I think because of https://github.com/grafana/grafana/blob/3d9a0c83982f4a0997fc44dbaab839248159629e/pkg/infra/tracing/tracing.go#L208. If I read the documentation seems like  `OTEL_SERVICE_NAME` only take precedence if `OTEL_RESOURCE_ATTRIBUTES` environment variables are used. Since we always set the service name in code no environment variables take precedence.

Reply:
> If I read the documentation seems like OTEL_SERVICE_NAME only take precedence if OTEL_RESOURCE_ATTRIBUTES environment variables are used

What that means is it's possible to set `service.name` via the `OTEL_RESOURCE_ATTRIBUTES` var, but `OTEL_SERVICE_NAME` should work regardless. It's likely due to how the OTel client is being configured.

Reply:
@hairyhenderson okay I've fixed it with https://github.com/grafana/grafana/pull/85715/commits/cbde5e1af6509032fd7c1aeedbda20e6f26d80ba. Now otel env variables take precedence over any set attributes. That's what you want?

Reply:
After consideration I'm going to open a separate PR for this change since it's changelog worthy/a feature.

Reply:
https://github.com/grafana/grafana/pull/85937